internal/sanitize_test.go

```diff
@@ -37,8 +37,18 @@ func TestSanitize(t *testing.T) {
 		},
 		{
 			name:  "don't modify alphanumeric string",
-			input: "0123456789_ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
-			want:  "0123456789_ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
+			input: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_0123456789",
```

Comment:
Perhaps this name also needs to change, as the precondition is now false.
We shall modify an alphanumeric string iff
the first key is either "_" or a digit

Maybe we can further update this to
```go
  {
    name: "don't modify alphanumeric string with letters as the prefix",
    input: " "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_0123456789",
    want: ""ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_0123456789",
},
```

Also is there a doc for StackDrive metric key sanitization?  I ask because for the past 3 days there have been many issues opened, each one coming with new rules so perhaps it's time for us to consolidate them in one place?

Reply:
I just named it "valid input". There is no doc that lists the limitations unfortunately :(